### PR TITLE
Return only the date part when JSON encoding Date/FrozenDate objects.

### PR DESCRIPTION
**Pull request by ADmad**
Refs #13170.

### CODE REVIEW

File: src/I18n/Date.php [hunk 1]
```diff
@@ -45,6 +45,22 @@ class Date extends MutableDate implements I18nDateTimeInterface
      */
     protected static $_toStringFormat = [IntlDateFormatter::SHORT, -1];
 
+    /**
+     * The format to use when when converting this object to json
```
Comment (dereuromark) [inline src/I18n/Date.php]: JSON is probably an acronym here.

Comment (ADmad) [reply src/I18n/Date.php]: I simply cut - pasted this from the `DateTimeTrait`. There's a double "when" too in the statement :)

Comment (saeideng) [reply src/I18n/Date.php]: `... when when ...`

**Comment by markstory** [comment]
Seems reasonable but also something that we'll need to document in the migration guide as it will change the output and behavior of people's public APIs.

**Comment by ADmad** [comment]
@markstory Already opened a PR for the migration guide to :)

**Comment by ADmad** [comment]
@markstory Good to merge?

**Comment by dereuromark** [comment]
He already merged docs one anyway :)

**Pull request merged**